hardware/serial/serial.cpp

```diff
@@ -2,11 +2,11 @@
 #include "stdafx.h"
 #include <algorithm>
 
-#if !defined(_WIN32) && !defined(__OpenBSD__) && !defined(__FreeBSD__)
+#if !defined(_WIN32) && !defined(__OpenBSD__) && !defined(__FreeBSD__) && !defined(__NetBSD__)
 # include <alloca.h>
 #endif
 
-#if defined (__MINGW32__)
+#if defined (__MINGW32__) || defined(__NetBSD__)
```

Comment:
This looks like a need for `-std=g++11` instead of redefining alloca.

Reply:
I did that (well, actually -std=c++11), but it still ends up saying, at link time:

CMakeFiles/domoticz.dir/hardware/serial/serial.cpp.o: In function `serial::Serial::readline(std::string&, unsigned long, std::string)':
serial.cpp:(.text+0x4c9): warning: Warning: reference to the libc supplied alloca(3); this most likely will not work. Please use the compiler provided version of alloca(3), by supplying the appropriate compiler flags (e.g. not -std=c89).

Reply:
Ah, I didn't know about gnu++11 - thanks!